big-red-shuttle/StopsViewController.swift

```diff
@@ -60,8 +62,43 @@ class StopsViewController: UIViewController, UITableViewDelegate, UITableViewDat
         let locations = stops.map { $0.getCoordinate() }
         setLocations(locations: locations)
         viewIsSetup = true
+        
+        //observe gps data changes
+        system().gps().addObserver(observer: self)
+        Location.shared.fetchedUserLocationCompletionBlock = {
+            if !system().gps().registeredToLogLocation {
+                system().api().registerUserToLogLocation(key: "fedorko", success: { (json: JSON?) in
+                    system().gps().startLoggingShuttleLocation()
+                    system().gps().startFetchingShuttleLocation()
+                }, failure: nil)
+            }
+        }
+    }
+    
+    //MARK: - Shuttle Bus GPS Delegate Protocol Methods
+    func gps(gps: GPS, movedToCoordinate coordinate: Coordinate) {
```

Comment:
I see, wasn't thinking it was a delegate call. I don't think delegation is the right move for a shared instance, should use either observation or notifications.